### PR TITLE
Fix 187

### LINKED ISSUE #187: Implement missing check for specific periods (e.g. season=djf)
At the moment the missing function checks for all dates in the `freq`, even though we might only be interested in a given month.

### PR DESCRIPTION
**Pull request by huard**
* **Please check if the PR fulfills these requirements**
- [x] This PR addresses an already opened issue (for bug fixes / features)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, etc.)
`missing_any` now handles season or month selectors.

* **Does this PR introduce a breaking change?** (Has there been an API change?)
I don't think so. 

Fixes #187

### CODE REVIEW

**Comment by huard** [comment]
Hold on, found issue.

File: tests/test_checks.py [hunk 1]
```diff
@@ -91,3 +91,29 @@ def test_to_period_end(self, tasmin_series):
...
+        miss = checks.missing_any(ts, freq='YS', month=[7, 8])
+        np.testing.assert_equal(miss, [False])
+
+    def test_season(self, tasmin_series):
+        ts = tasmin_series(np.zeros(360))
+        miss = checks.missing_any(ts, freq='YS', season='MAM')
+        np.testing.assert_equal(miss, [False])
+
+        miss = checks.missing_any(ts, freq='YS', season='JJA')
+        np.testing.assert_array_equal(miss, [True, True])
```
Comment (tlvu) [inline tests/test_checks.py]: Not sure but why not just use `assert_array_equals` everywhere?  An array of 1 element is still an array.

Comment (huard) [reply tests/test_checks.py]: Correct, that would work, just a bit more verbose.

**Comment by huard** [comment]
@tlogan2000 Ok if I merge this ?

**Comment by tlogan2000** [comment]
Yes .. sorry I got distracted last week when I was looking at this

**Pull request merged**